vnext/PropertySheets/ReactPackageDirectories.props

```diff
@@ -13,22 +13,25 @@
 
   <PropertyGroup>
     <ReactNativeWindowsDir Condition="'$(ReactNativeWindowsDir)' == ''">$(MSBuildThisFileDirectory)\..\</ReactNativeWindowsDir>
-    <ReactNativeDir Condition="'$(ReactNativeDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'node_modules\react-native\package.json'))\node_modules\react-native\</ReactNativeDir>
+    <ReactNativePackageDir Condition="'$(ReactNativePackageDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'node_modules\react-native\package.json'))\node_modules\react-native\</ReactNativePackageDir>
+    <ReactNativeDir Condition="'$(IntDir)' != ''">$(IntDir)\react-native-patched\</ReactNativeDir>
+    <ReactNativeDir Condition="'$(IntDir)' == ''">$(BaseIntermediateOutputPath)\react-native-patched\</ReactNativeDir>
     <YogaDir Condition="'$(YogaDir)' == ''">$(ReactNativeDir)\ReactCommon\yoga</YogaDir>
   </PropertyGroup>
 
-  <PropertyGroup Condition="!Exists('$(ReactNativeDir)\stubs\glog\logging.h')">
+  <PropertyGroup Condition="!Exists('$(ReactNativePackageDir)\stubs\glog\logging.h')">
     <OSS_RN>true</OSS_RN>
```

Comment:
Reminder to myself to add a comment here, that this logic is used to examine the package, determine if we are upstream, then set a property so we can tell if we're using upstream using a preprocessor definition. I was wondering how we determined that before, and was a bit confused by this logic when I first saw it.